### ISSUE TITLE
[edge-mdns]: Cannot resolve address when broacasting with service. 

### ISSUE THREAD

**Issue by AnthonyGrondin** [ivmarkov/edge-net#25]
When using edge-mdns as a responder while broadcasting a service using the `ServiceAnswers` wrapper, it fails to reply to host queries. This is while running with `edge-nal-embassy` on esp32s3 with `esp-wifi`, but I'd assume it has the same behavior on std.

Example of code:
```rust
async fn run<T, RB, SB>(
    stack: &T,
    recv_buf: RB,
    send_buf: SB,
    our_name: &str,
    our_ip: core::net::Ipv4Addr,
) -> Result<(), MdnsIoError<T::Error>>
where
    T: UdpBind,
    RB: BufferAccess<[u8]>,
    SB: BufferAccess<[u8]>,
{
    let mut socket = edge_mdns::io::bind(
        stack,
        DEFAULT_SOCKET,
        Some(core::net::Ipv4Addr::UNSPECIFIED),
        None,
    )
    .await?;

    let (recv, send) = socket.split();

    let host = Host {
        hostname: our_name,
        ipv4: our_ip,
        ipv6: Ipv6Addr::UNSPECIFIED,
        ttl: Ttl::from_secs(60),
    };

    let service = Service {
        name: "my-service",
        priority: 1,
        weight: 5,
        service: "_https",
        protocol: "_tcp",
        port: 443,
        service_subtypes: &[],
        txt_kvs: &[]
    };

    let signal = Signal::new();

    let mdns = edge_mdns::io::Mdns::<NoopRawMutex, _, _, _, _>::new(
        Some(core::net::Ipv4Addr::UNSPECIFIED),
        None,
        recv,
        send,
        recv_buf,
        send_buf,
        |buf| {},
        &signal,
    );

    mdns.run(HostAnswersMdnsHandler::new(ServiceAnswers::new(&host, &service))).await
```

That same code works if we remove the `ServiceAnswers` wrapper to only handle `host`.

**Comment by ivmarkov** [comment]
What exactly does the failure look like?
Panic? Or something else?

**Comment by AnthonyGrondin** [comment]
It fails to resolve the address on the client side. I'm testing with avahi on linux, and when I do:
```bash
avahi-resolve-host-name hostname.local
Failed to resolve host name 'hostname.local': Timeout reached
```
It times out and the following is logged on the esp32:
```
INFO - Replying to mDNS query from 192.168.69.69:5353
INFO - Broadcasting mDNS entry to 224.0.0.251:5353
INFO - Replying to mDNS query from 192.168.69.69:5353
INFO - Broadcasting mDNS entry to 224.0.0.251:5353
INFO - Replying to mDNS query from 192.168.69.69:5353
INFO - Broadcasting mDNS entry to 224.0.0.251:5353
```

When I try to resolve the service, it successfully resolves the service, but avahi tries to resolves the host details too:
```bash
avahi-browse -rt _https._tcp
+ enp4s0f0 IPv4 my-service                                         Secure Web Site      local
Failed to resolve service 'my-service' of type '_https._tcp' in domain 'local': Timeout reached
```

**Comment by ivmarkov** [comment]
Let me do a quick check...

**Comment by ivmarkov** [comment]
You are right! [I have forgotten one line in `ServiceAnswers`](https://github.com/ivmarkov/edge-net/commit/406007815ac213d69d68615f0e0c51cd912f8acc) :-(

Can you do `cargo update` and try again? I just fixed and re-published `edge-mdns 0.3.1` as well as `edge-net 0.8.1`.

**Comment by AnthonyGrondin** [comment]
Thank you for taking care of this. I can confirm it solves the issue I was having, and I can now successfully resolve hostnames and services.
